--- origsrc/luaxmlrpc-1.0b/cgi.lua	2004-12-02 12:32:08.000000000 -0600
+++ src/luaxmlrpc-1.0b/cgi.lua	2008-02-13 15:15:59.906250000 -0600
@@ -2,7 +2,7 @@
 -- $Id: lua-xmlrpc-1.0b-1.src.patch,v 1.1 2008-02-13 21:18:06 yselkowitz Exp $
 
 require"xmlrpc"
-require"post"
+require"cgilua.post"
 post.setmaxinput (1024 * 1024)
 post.setmaxfilesize (512 * 1024)
 
--- origsrc/luaxmlrpc-1.0b/http.lua	2004-12-02 12:32:08.000000000 -0600
+++ src/luaxmlrpc-1.0b/http.lua	2008-02-13 15:11:45.921875000 -0600
@@ -4,11 +4,10 @@
 -- $Id: lua-xmlrpc-1.0b-1.src.patch,v 1.1 2008-02-13 21:18:06 yselkowitz Exp $
 ---------------------------------------------------------------------
 
-require"socket"
+require"socket.http"
+require"ltn12"
 require"xmlrpc"
 
-local post = socket.http.post
-
 xmlrpc.http = {}
 
 ---------------------------------------------------------------------
@@ -19,17 +18,22 @@
 --	XML-RPC element).
 ---------------------------------------------------------------------
 function xmlrpc.http.call (url, method, ...)
-	local body, headers, code, err = post {
+	local body = {}
+	local data = xmlrpc.clEncode (method, unpack (arg))
+	local rc, code, headers = socket.http.request {
+		method = "POST",
+		source = ltn12.source.string(data),
 		url = url,
-		body = xmlrpc.clEncode (method, unpack (arg)),
 		headers = {
 			["User-agent"] = "LuaXMLRPC",
 			["Content-type"] = "text/xml",
+			["Content-length"] = tostring(string.len(data)),
 		},
 	}
-	if tonumber (code) == 200 then
+	body = table.concat(body)
+	if rc == 1 and tonumber (code) == 200 then
 		return xmlrpc.clDecode (body)
 	else
-		error (err or code)
+		error (code)
 	end
 end
--- origsrc/luaxmlrpc-1.0b/xmlrpc.lua	2004-12-02 12:32:08.000000000 -0600
+++ src/luaxmlrpc-1.0b/xmlrpc.lua	2008-02-13 15:06:31.328125000 -0600
@@ -13,7 +13,7 @@
 local ceil = math.ceil
 local parse = lxp.lom.parse
 
-module (arg and arg[1])
+module("xmlrpc")
 
 _COPYRIGHT = "Copyright (C) 2003-2004 Kepler Project"
 _DESCRIPTION = "LuaXMLRPC is a library to make remote procedure calls using XML-RPC"
